### ISSUE TITLE
[iOS] cannot setValue on AnimatedValue which has been moved to native driven animation 

### ISSUE THREAD

**Issue by magicismight** [facebook/react-native#10015]
How to reproduce this:

First use this value on a native driven animation:

```
const value = new Animated.Value(0);

Animated.timing(value, {
  useNativeDriver: true,
  ...other animation configure
});
```

Then call setValue, and this change won't  be updated on the AnimatedNode.
Android works fine, only appears on iOS.

```
value.setValue(someValue);
```
- React Native version: 0.33.0
- Platform(s) (iOS, Android, or both?): iOS
- Operating System (macOS, Linux, or Windows?):macOS

**Comment by JakeRawr** [comment]
Adding onto this which I believe it's the same issue
The animated api only animate once with `useNativeDriver: true`

example:

```
...
_onPress() {
  Animated.timing(this._animatedValue, {
      toValue: 100,
      duration: 3600,
      easing: Easing.in(Easing.bezier(0,0,0,1)),
      useNativeDriver: true
    }).start()
}

...
render() {
  var interpolatedRotateAnimation = this._animatedValue.interpolate({
      inputRange: [0, 100],
      outputRange:['0deg', '360deg']
    });

  return (
    <Animated.View
        style={
        [styles.box, {transform: [{rotate: interpolatedRotateAnimation}]}]
      }>
          <Text onPress={this._onPress}>{'Press to Turn'}</Text>
     </Animated.View>
  )
}
...
```
The above example only spin the component *once*, pressing it second time does not do the animation. 

If I unmount the entire Component and come back, it would work *once* again, tho.

React Native version: 0.34.0
Platform(s) (iOS, Android, or both?): iOS
Operating System (macOS, Linux, or Windows?):macOS

**Comment by JakeRawr** [comment]
Was able to fix it by resetting `this._animatedValue` in `.start()`

example below:

```
...
_onPress() {
  Animated.timing(this._animatedValue, {
      toValue: 100,
      duration: 3600,
      easing: Easing.in(Easing.bezier(0,0,0,1)),
      useNativeDriver: true
    }).start( () => {
       this._animatedValue = new Animated.Value(0)
    })
}

...
render() {
  var interpolatedRotateAnimation = this._animatedValue.interpolate({
      inputRange: [0, 100],
      outputRange:['0deg', '360deg']
    });

  return (
    <Animated.View
        style={
        [styles.box, {transform: [{rotate: interpolatedRotateAnimation}]}]
      }>
          <Text onPress={this._onPress}>{'Press to Turn'}</Text>
     </Animated.View>
  )
}
...
```

However, I have not tested if there's a memory leak or if there's any other possible toll I have to pay. If anyone see a problem with this approach, let me know.

**Comment by magicismight** [comment]
This has been fixed in #10643
